annotatedcsv/decode.go

```diff
@@ -0,0 +1,451 @@
+// Copyright 2021 InfluxData, Inc. All rights reserved.
+// Use of this source code is governed by MIT
+// license that can be found in the LICENSE file.
+
+package annotatedcsv
+
+import (
+	"encoding/base64"
+	"fmt"
+	"reflect"
+	"strconv"
+	"time"
+
+	ireflect "github.com/influxdata/influxdb-client-go/internal/reflect"
+)
+
+// Decode decodes the current row into x, which should be
+// a pointer to a struct or a pointer to a slice.
+//
+// When it's a pointer to a struct, columns in the row are decoded into
+// appropriate fields in the struct, using the similar tag conventions
+// described by encoding/json to determine how to map
+// column names to struct fields. Tag prefix must start with "flux:":
+//
+//  type Point struct {
+//      Timestamp  time.Time `flux:"_time"`
```

Comment:
I'm wondering if `flux` is the correct annotation here, and thinking about `csv` instead, as we're really about CSV here, and flux might have several possible output formats in the future. WDYT?

Reply:
`flux` sounds, indeed, specific to the Flux results. As annotatedcsv is package for generally annotated CSV data, so `csv` is better.